membership/src/main/scala/zio/membership/transport/Transport.scala

```diff
@@ -0,0 +1,26 @@
+package zio.membership.transport
+
+import zio._
+import zio.stream._
+import zio.nio.SocketAddress
+
+trait Transport {
+  val transport: Transport.Service[Any]
+}
+
+object Transport {
+
+  /**
+   * TODO: Naming should be discussed.
+   * Our low level transport interface that allows sending messages either in fire-and-forget
+   * or in streaming style. Also allows listening to messages sends from other nodes.
+   *
+   * The default implementation of this should use both TCP and UDP and bind both protocols
+   * to the same port number on bind.
+   */
+  trait Service[R] {
+    def sendBestEffort(to: SocketAddress, msg: Chunk[Byte]): ZIO[R, Error, Unit]
```

Comment:
why do you think that we should distinct this in here? why not send send/bind only?

Reply:
Performance mostly. I believe as much as possible should be done in udp for performance reasons, but for example user messages or large messages might be better send using tcp.

Also for example consul uses both in case a node is not reachable using udp but with tcp.

I'm not sure how exactly we want to do this, but I believe we should make both available.

Reply:
Good reasons. question is more about if this belongs to Transport. I thought that Transport is just ```UDP``` or ```TCP```. In your example with consul I would use both, not one combined.

I was thinking about something similar to this https://haskell-distributed.github.io/tutorials/1ch.html#sending-messages

`
import Network.Transport.TCP (createTransport, defaultTCPParameters)
....

main = do
  Right t <- createTransport "127.0.0.1" "10501" defaultTCPParameters
...
`

Reply:
One issue  I was thinking about is that some algorithms specifically need and reliable transport layer for failure detection.
This gets us to OOP land, but would do you think about having Transport <:< UnreliableTransport?

Reply:
your are optimizing for something that is not implemented or even planned. It would be better to just always keep MVP approach that we deliver minimal thing.